src/main/java/edu/ucla/library/avpairtree/verticles/ConverterVerticle.java

```diff
@@ -0,0 +1,159 @@
+     */
+    private static final String DEFAULT_ENCODING_FORMAT = "mp4";
+
+    @Override
+    public void start(final Promise<Void> aPromise) {
+        final DeliveryOptions options = new DeliveryOptions().setSendTimeout(Integer.MAX_VALUE);
+        final JsonObject config = config();
+        final String sourceDir = config.getString(Config.SOURCE_DIR);
+        final Vertx vertx = getVertx();
+
+        LOGGER.debug(MessageCodes.AVPT_011, ConverterVerticle.class.getSimpleName(), Thread.currentThread().getName());
+
+        vertx.eventBus().<CsvItem>consumer(getClass().getName()).handler(message -> {
+            final String outputFormat = config.getString(Config.ENCODING_FORMAT, DEFAULT_ENCODING_FORMAT);
+            final CsvItem csvItem = message.body();
+
+            try {
+                final Path inputFilePath = getInputFilePath(csvItem, sourceDir);
+                final Path outputFilePath = getOutputFilePath(inputFilePath, outputFormat);
+                final EncodingAttributes encoding = new EncodingAttributes();
+                final AudioAttributes audio = new AudioAttributes();
+                final Encoder encoder = new Encoder();
+
+                audio.setCodec(config.getString(Config.AUDIO_CODEC, DEFAULT_AUDIO_CODEC));
+                audio.setBitRate(config.getInteger(Config.BIT_RATE, DEFAULT_BIT_RATE));
+                audio.setChannels(config.getInteger(Config.CHANNELS, DEFAULT_CHANNELS));
+                audio.setSamplingRate(config.getInteger(Config.SAMPLING_RATE, DEFAULT_SAMPLING_RATE));
+
+                encoding.setOutputFormat(outputFormat);
+                encoding.setAudioAttributes(audio);
+
+                encoder.encode(new MultimediaObject(inputFilePath.toFile()), outputFilePath.toFile(), encoding);
+                csvItem.setFilePath(outputFilePath.toAbsolutePath().toString());
+
+                // Send our converted file to the Pairtree verticle for placement in the A/V Pairtree
+                vertx.eventBus().request(PairtreeVerticle.class.getName(), csvItem, options).onSuccess(result -> {
+                    // Clean up our converted file after it has been successfully put into the Pairtree
+                    vertx.fileSystem().delete(outputFilePath.toAbsolutePath().toString()).onComplete(deletion -> {
+                        if (deletion.succeeded()) {
+                            // If our scratch space file was cleaned up, report the success back to the watcher
+                            message.reply(result.body());
+                        } else {
+                            LOGGER.error(deletion.cause(), deletion.cause().getMessage());
+                            message.fail(Op.ERROR_CODE, deletion.cause().getMessage());
+                        }
+                    });
+                }).onFailure(error -> {
+                    LOGGER.error(error, error.getMessage());
+                    message.fail(Op.ERROR_CODE, error.getMessage());
+                });
```

Comment:
So I understand... you're proposing that the ConverterVerticle converts and send the response back to the WatcherVerticle and then the WatcherVerticle sends the request to the PairtreeVerticle?

Reply:
yes

Reply:
I'm torn on this. I see what you're saying, but it's also taking a longer, perhaps more convoluted path, to achieve the goal (rather than taking the shortest path between two points). It's not a hard change to make, though, so if you feel it's better I'll make it.

Reply:
I'm not clear on how it would make the path longer or more convoluted, since there's the same number of send/replies with each approach, it's just that in the approach I'm suggesting, WatcherVerticle initiates both of them. Wouldn't the code look more or less the same if it were in WatcherVerticle instead?

And I'm not sure about principles for event-driven systems specifically, but I think regardless of programming paradigm, reducing dependencies and side effects (or at least, reducing the _number of places_ where side effects occur, i.e. put all the calls to PairtreeVerticle near each other) are always good things to strive for.

Reply:
There is an extra message being sent by sending a message back to the WatcherVerticle and having it make the request to the PairtreeVerticle (vs. going directly from the ConverterVerticle to the PairtreeVerticle)... so, four hops instead of three.

I agree reducing dependencies of an application is a good thing. Viewing this particular change as a dependency change though seems a bit forced to me. I think I'm understanding what you're arguing... just not sure I see it the same way.

I don't really have a problem with doing it, though, so if you feel more strongly about it than I do it will get done. :-)

Reply:
I think I'm not understanding the argument. Will chat via Slack to get a better grasp.

Reply:
To quickly summarize the discussion: Mark feels using a spoke model, where communication goes out of and back into a centralized controller makes the workflow easier to grok. I like the more distributed communication between verticles on the event queue and think something like AsyncAPI would make the workflow easier to grok (though for this simple app I also don't think it's hard to grok). For this particular application, I don't think sending all the messages through a centralized point is going to have a huge performance penalty, so I'll apply the change; if we have problems with it in the future, we can test each approach to see which performs better. I'll keep it as a separate commit in the history.

Reply:
Implemented, so resolving this conversation.

Reply:
After discussing more, I no longer feel that my "improvement" suggestion is actually an improvement; the need to delete converted mp4 files (and the question of where to do that) is something I hadn't considered. I think my suggestion simplifies things in some areas but complicates them in others, so please disregard

Reply:
Okay, rolled back the newer commits.